rpc/contract_test.go

```diff
@@ -202,47 +232,61 @@ func TestClass(t *testing.T) {
+				ExpectedProgram:               utils.TestHexToFelt(t, "0x576402000a0028a9c00a010").String(),
+				ExpectedEntryPointConstructor: SierraEntryPoint{FunctionIdx: 34, Selector: utils.TestHexToFelt(t, "0x28ffe4ff0f226a9107253e17a904099aa4f63a02a5621de0576e5aa71bc5194")},
+			},
 			{
-				BlockID:                       WithBlockHash(utils.TestHexToFelt(t, "0x464fc8c86a6452536a2e27cb301815e5f8b16a2f6872ba4f3d83701fbe99fb3")),
-				ClassHash:                     utils.TestHexToFelt(t, "0x011fbe1adeb2afdf5b545f583f8b5a64fb35905f987d249193ad8185f6fcf571"),
-				ExpectedEntryPointConstructor: SierraEntryPoint{FunctionIdx: 16, Selector: utils.TestHexToFelt(t, "0x28ffe4ff0f226a9107253e17a904099aa4f63a02a5621de0576e5aa71bc5194")},
+				BlockID:                       WithBlockTag("latest"),
+				ClassHash:                     utils.TestHexToFelt(t, "0x01f372292df22d28f2d4c5798734421afe9596e6a566b8bc9b7b50e26521b855"),
+				ExpectedProgram:               utils.TestHexToFelt(t, "0xe70d09071117174f17170d4fe60d09071117").String(),
+				ExpectedEntryPointConstructor: SierraEntryPoint{FunctionIdx: 2, Selector: utils.TestHexToFelt(t, "0x28ffe4ff0f226a9107253e17a904099aa4f63a02a5621de0576e5aa71bc5194")},
+			},
+		},
+		"mainnet": {
+			// v2 class
+			{
+				BlockID:                       WithBlockTag("latest"),
+				ClassHash:                     utils.TestHexToFelt(t, "0x029927c8af6bccf3f6fda035981e765a7bdbf18a2dc0d630494f8758aa908e2b"),
+				ExpectedProgram:               utils.TestHexToFelt(t, "0x9fa00900700e00712e12500712e").String(),
+				ExpectedEntryPointConstructor: SierraEntryPoint{FunctionIdx: 32, Selector: utils.TestHexToFelt(t, "0x28ffe4ff0f226a9107253e17a904099aa4f63a02a5621de0576e5aa71bc5194")},
 			},
 		},
-		"mainnet": {},
 	}[testEnv]
 
 	for _, test := range testSet {
+		require := require.New(t)
 		spy := NewSpy(testConfig.provider.c)
 		testConfig.provider.c = spy
-		resp, err := testConfig.provider.Class(context.Background(), WithBlockTag("latest"), test.ClassHash)
-		if err != nil {
-			t.Fatal(err)
-		}
+		resp, err := testConfig.provider.Class(context.Background(), test.BlockID, test.ClassHash)
+		require.NoError(err)
 
 		switch class := resp.(type) {
-		case DeprecatedContractClass:
-
+		case *DeprecatedContractClass:
 			diff, err := spy.Compare(class, false)
-			if err != nil {
-				t.Fatal("expecting to match", err)
-			}
-			if diff != "FullMatch" {
-				if _, err := spy.Compare(class, true); err != nil {
-					log.Fatal(err)
-				}
-				t.Fatal("structure expecting to be FullMatch, instead", diff)
-			}
+			require.NoError(err, "expecting to match")
+			require.Equal(diff, "FullMatch", "structure expecting to be FullMatch")
 
 			if !strings.HasPrefix(class.Program, test.ExpectedProgram) {
 				t.Fatal("code should exist")
 			}
-		case ContractClass:
-			require.Equal(t, class.EntryPointsByType.Constructor, test.ExpectedEntryPointConstructor)
+		case *ContractClass:
+			diff, err := spy.Compare(class, false)
```

Comment:
Is thi spy doing anything? can we remove it?

Reply:
Agree. Done